### PR TITLE
Fix structures of `never` not considered as subtypes of `anydata`

### LINKED ISSUE #30141: Structures of `never` aren't considered as subtypes of `anydata`
**Description:**
$title.

**Steps to reproduce:**
```ballerina
public function main() {
    map<never> mn = {};
    never[] ar = [];

    // Compilation errors.
    anydata a = mn; // incompatible types: expected 'anydata', found 'map<never>'
    anydata b = ar; // incompatible types: expected 'anydata', found 'never[]'

    // Allowed
    any c = mn;    
    any d = ar;  

    json e = mn;  
    json f = ar;  
}
```

**Affected Versions:**
slalpha5-SNAPSHOT

### PR DESCRIPTION
**Pull request by lasinicl**
## Purpose
$title

Fixes #30141

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-unit-test/src/test/resources/test-src/types/never/never-type.bal [hunk 1]
```diff
@@ -425,6 +425,25 @@ function bar() returns record {| never x; |} {
...
 
+function testNeverWithAnydata() {
+    map<never> a = {};
+    never[] b = [];
+
+    anydata m = a;
+    anydata n = b;
+
+    anydata x1 = baz1();
+    anydata x2 = baz2();
```
Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/resources/test-src/types/never/never-type.bal:436]: Shall we also use these in assertions? 

e.g.,
```ballerina
assertEquality(true, m is map<anydata>);
map<anydata> mp = <map<anydata>> m;
assertEquality(0, mp.length());
```

Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/resources/test-src/types/never/never-type.bal:436]: Please add runtime tests too.

e.g.,
```ballerina
any v = a;
assertEquality(true, v is anydata);
```

File: tests/jballerina-unit-test/src/test/resources/test-src/types/never/never-type-runtime.bal [hunk 2]
```diff
@@ -78,6 +78,39 @@ function test6() {
     panic error("Error value is:" + string `${b}`);
 }
 
+function testNeverWithAnyAndAnydataRuntime() {
+    map<never> a = {};
+    never[] b = [];
+
+    anydata m = a;
+    assertEquality(true, m is map<any>);
+    assertEquality(true, m is any);
```
Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/resources/test-src/types/never/never-type-runtime.bal:87]: Shall we assign `a` to an `any` variable and specifically check `is anydata` also?

Comment (lasinicl) [reply tests/jballerina-unit-test/src/test/resources/test-src/types/never/never-type-runtime.bal:87]: Fixed in b9d9424 
Came across https://github.com/ballerina-platform/ballerina-lang/issues/30264

**Review by MaryamZi — APPROVED** [review]

**Pull request merged**